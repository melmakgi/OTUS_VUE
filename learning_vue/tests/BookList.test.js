import {describe, it, expect} from "vitest";
import Component from "@/components/BookItem2.vue";
import {mount} from '@vue/test-utils'

const items = [{id: 1}, {id: 2}, {id: 3}, {id: 4}, {id: 5}];
const wrapper = mount(Component, {props: {items}});

describe('Component BookItem2', () => {
    it('Mounts without errors', () => {
        expect(wrapper).toBeDefined();
        expect(wrapper.exists()).toBe(true);   // Проверка: нашли или нет
    })
    it('Renders list', () => {
        expect(wrapper.html()).toContain({id: 1});
        expect(wrapper.html()).toContain({id: 2});
        expect(wrapper.html()).toContain({id: 3});
        expect(wrapper.html()).toContain({id: 4});
        expect(wrapper.html()).toContain({id: 5});
    })
    it('Renders All', () => {
        expect(wrapper.html()).toMatchSnapshot();
    })
    it('Emits event', () => {
        const removeControl = wrapper.find("button");
        // Проверка: нашли или нет
        expect(removeControl.exists()).toBe(true);
        removeControl.trigger('click');
        // проверка, что событие было вызвано
        expect(wrapper.emitted()).toBeTruthy()
        //console.log(wrapper.emitted());
        expect(wrapper.emitted().remove);
    })
})

