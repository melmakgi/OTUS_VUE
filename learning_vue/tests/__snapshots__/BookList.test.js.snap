// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Component > Renders All 1`] = `
"<div class=\\"row g-0\\" items=\\"[object Object],[object Object],[object Object],[object Object],[object Object]\\">
  <div class=\\"col-md-4\\"><img src=\\"/images/undefined\\" class=\\"img-fluid rounded-start\\"></div>
  <div class=\\"col-md-8\\">
    <div class=\\"card-body\\">
      <h5 class=\\"card-title mb-3\\"></h5>
      <h6 class=\\"card-title mb-3\\"></h6>
      <p class=\\"card-text text-truncate\\"> Описание: </p>
      <p class=\\"card-text\\">Категория: </p>
      <p class=\\"card-text\\"> Издательство: . Год: </p>
      <p class=\\"card-text\\">Цена: руб.</p>
      <p class=\\"card-text\\"><small class=\\"text-muted\\">ISBN: </small></p><button type=\\"button\\" class=\\"btn btn-primary\\" mt-3=\\"\\" mb-3=\\"\\">Удалить</button>
    </div>
  </div>
</div>"
`;

exports[`Component BookItem2 > Renders All 1`] = `
"<div class=\\"row g-0\\" items=\\"[object Object],[object Object],[object Object],[object Object],[object Object]\\">
  <div class=\\"col-md-4\\"><img src=\\"/images/undefined\\" class=\\"img-fluid rounded-start\\"></div>
  <div class=\\"col-md-8\\">
    <div class=\\"card-body\\">
      <h5 class=\\"card-title mb-3\\"></h5>
      <h6 class=\\"card-title mb-3\\"></h6>
      <p class=\\"card-text text-truncate\\"> Описание: </p>
      <p class=\\"card-text\\">Категория: </p>
      <p class=\\"card-text\\"> Издательство: . Год: </p>
      <p class=\\"card-text\\">Цена: руб.</p>
      <p class=\\"card-text\\"><small class=\\"text-muted\\">ISBN: </small></p><button type=\\"button\\" class=\\"btn btn-primary\\" mt-3=\\"\\" mb-3=\\"\\">Удалить</button>
    </div>
  </div>
</div>"
`;
